ReactWindows/ReactNative/Modules/Location/LocationModule.cs

```diff
@@ -0,0 +1,194 @@
+                    StatusChangedEventArgs>(
+                        h => geolocator.StatusChanged += h,
+                        h => geolocator.StatusChanged -= h);
+
+            var statusQuery = from e in statusObservable
+                              select e.EventArgs.Status;
+
+            var statusSubscription = statusQuery.Subscribe(status =>
+            {
+                if (status == PositionStatus.Disabled)
+                {
+                    Context.GetJavaScriptModule<RCTDeviceEventEmitter>()
+                        .emit("geolocationError", "Location settings are turned off.");
+                }
+                else if (status == PositionStatus.NotAvailable)
+                {
+                    Context.GetJavaScriptModule<RCTDeviceEventEmitter>()
+                        .emit("geolocationError", "Location services is not available.");
+                }
+                else if (status == PositionStatus.NoData || status == PositionStatus.Initializing)
+                {
+                    Context.GetJavaScriptModule<RCTDeviceEventEmitter>()
+                        .emit("geolocationError", "Location services is temporarily unavailable.");
+                }
+            });
+
+            _geolocatorSubscription = new CompositeDisposable(2)
+            {
+                positionSubscription,
+                statusSubscription,
+            };
+        }
+
+        [ReactMethod]
+        public void stopObserving()
+        {
+            if (_geolocatorSubscription != null)
+            {
+                _geolocatorSubscription.Dispose();
+                _geolocatorSubscription = null;
+            }
+        }
+
+        private static JObject ConvertGeoposition(Geoposition geoposition)
+        {
+            var coords = new JObject
+            {
+                { "latitude", geoposition.Coordinate.Point.Position.Latitude },
+                { "longitude", geoposition.Coordinate.Point.Position.Longitude },
+                { "altitude", geoposition.Coordinate.Point.Position.Altitude },
+                { "accuracy", geoposition.Coordinate.Accuracy },
+                { "bearing", geoposition.Coordinate.Heading },
+                { "speed", geoposition.Coordinate.Speed }
+            };
+
+            return new JObject
+            {
+                { "coords", coords },
+                { "timestamp", geoposition.Coordinate.Timestamp.ToUnixTimeMilliseconds() },
+            };
+        }
+
+        class LocationOptions
+        {
+            private const double DefaultLocationAccuracy = 100;
+
+            private LocationOptions(long timeout, double maximumAge, bool highAccuracy, double distanceFilter)
+            {
+                Timeout = timeout;
+                MaximumAge = maximumAge;
+                HighAccuracy = highAccuracy;
+                DistanceFilter = distanceFilter;
+            }
+
+            public long Timeout { get; }
+
+            public double MaximumAge { get; }
```

Comment:
GetGeopositionHistoryAsync is throwing "AccessDenied", tried various things (like Geolocator.RequestAccessAsync) to no avail.

---
In reply to: [70985996](https://github.com/ReactWindows/react-native-windows/pull/526#discussion_r70985996) [](ancestors = 70985996)